### DISCUSSION TITLE
Specify the name of the heat source when assign openings

### DISCUSSION THREAD

**Discussion by aknsj** [ansys/pyaedt#5059] [Q&A] [answered]
Hi all, pyaedt is an effective automation tool that has helped me save a lot of time! 
But I recently ran into a small problem.I used the following code to assign openings for the faces of the air region with pyaedt.
```
ipk.assign_openings(air_faces=region.faces)
```
The following figure shows the properties of the opening heat source generated. The name is “Opening_QE2336”, which seems to consist of the prefix “Opening” and “_” and a randomly generated suffix consisting of letters and numbers.
![image](https://github.com/user-attachments/assets/188d5147-a6cd-448f-996e-b0899e9fef46)
My requirement is to set the specified name for the opening source or get the name of the generated opening boundary. The reason is that when I use the following code to output summary report，the parameter boundary_list of this function is a list containing the names of boundaries.
```
def create_summary_report_heatflowrate(boundary_list,csv_name):
    if not csv_name.endswith(".csv"):
        raise ValueError("csv name must end with '.csv'.")
    heatflowrate_fs = ipk.post.create_field_summary()
    heatflowrate_fs.add_calculation(
            entity="Boundary",
            geometry="Surface",
            geometry_name=boundary_list,
            quantity="HeatFlowRate",
            side="Combined",
            mesh="All",
        )
    heatflowrate_fs.export_csv(output_file=os.path.join(base_path,csv_name))
```
I'd love to know how to specify or get the name of the opening boundary.The version of pyaedt I use is 0.9.10.Looking forward to your reply. Thank you!

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @aknsj,

Almost all PyAEDT methods return an object. So in this case you can save the object in a variable:

opening = ipk.assign_openings(air_faces=region.faces)

And you can get the name and even set the name:

name = opening.name

opening.name = "samuel"

**Reply by aknsj** [reply]
Thank you very much, your answer helped me.
